### ISSUE TITLE
Wrong behaviour DT[, i, with=FALSE]  

### ISSUE THREAD

**Issue by cguill95** [Rdatatable/data.table#2109]
Hi,

I noticed a strange type of behaviour using the ```with=FALSE``` parameter in a data table. 
Maybe worth mentioning I am using data table version 1.10.4. on R 3.3.2

Here is a reproducible example;

    library(data.table)
    test <- data.table(mtcars)
    i <- 3L
    
    # Print 3rd column
    test[, i, with=FALSE]
    
    # Print 4th column
    test[, i + 1L, with=FALSE]
    
    # Print all but 3rd column
    test[, i - 1L, with=FALSE]

**Comment by franknarf1** [comment]
Yeah, I guess it parses the expression, sees that `-` is the top-level function and wings it from there somehow, similar to this bug: https://github.com/Rdatatable/data.table/issues/2069 Parentheses, stop it: `test[, (i - 1L), with=FALSE]`

**Comment by arunsrinivasan** [comment]
IIUC [this line](https://github.com/Rdatatable/data.table/blob/master/R/data.table.R#L903) would need a `length(jsub) == 2L` extra condition...
